specification.md

```diff
@@ -264,9 +264,14 @@ The state data filters `input` property expression is applied when the workflow
 It can be used to select only data that is needed and disregard what is not needed.
 If `input` is not defined or does not select any parts of the state's data input, its data input is not filtered.
 
-The state data filter `output` property expression is applied right before the state transitions to the next state defined. It filters the state's data output to be passed as data input to the transitioning state.
+The state data filter `output` property expression is applied right before the state transitions to the next state defined. 
+It filters the state's data output to be passed as data input to the transitioning state.
 If the current state is the workflow end state, the filtered state's data output becomes the workflow data output.
 If `output` is not defined or does not select any parts of the state's data output, its data output is not filtered.
+
+The results of the `input` and `output` expressions should be merged into the state data. 
```

Comment:
If this section is talking about the stateDataFilter and about the input/output, why would this be a 'merge'?  Also, shouldn't the result of `input` just become the state data (not a merge)?  And the result of `output`, doesn't that become the workflow data (also, not a merge)?

From another thread:

> the main driver to keep the workflow data as small as possible (for performance) is the state filters

This implies to me that `output` becomes the state data, no merging, otherwise again we're just accumulating data and never allowing the user to trim the data.

Reply:
@tsurdilo Not sure if you saw this.  This line is confusing because it implies merging where, based other statements, I believe there should not be merging.

Reply:
@jorgenj ok must have missed it (lots of comments :) ) Will address it asap

Reply:
+1 will update. good points

Reply:
done